sys/powerpc/ps3/ps3disk.c

```diff
@@ -588,13 +588,17 @@ ps3disk_transfer(void *arg, bus_dma_segment_t *segs, int nsegs, int error)
 	struct bio *bp = (struct bio *)(arg);
 	struct ps3disk_softc *sc = (struct ps3disk_softc *)bp->bio_disk->d_drv1;
 	struct ps3disk_region *rp = &sc->sc_reg[bp->bio_disk->d_unit];
+	bus_dma_segment_t *seg = &segs[0];
 	uint64_t devid = ps3bus_get_device(sc->sc_dev);
-	uint64_t block;
+	uint64_t block, bio_length, sector_op_count;
 	int i, err;
 
 	/* Locks already held by busdma */
 	PS3DISK_ASSERT_LOCKED(sc);
 
+	KASSERT(error == 0,
+		    ("DMA Error %d", error));
+
 	if (error) {
 		bp->bio_error = error;
 		bp->bio_flags |= BIO_ERROR;
```

Comment:
Does the KASSERT() before normal error recovery makes some sense, or it is a leak of debug code?

Reply:
Latter, to me, this error should never hit at anytime.

should i remove?

Reply:
I'm fine removing that.